asset/src/job_metric_api/api.ts

```diff
@@ -0,0 +1,208 @@
+        };
+        this.prefix = 'teraslice_job_';
+        this.setPodName();
+    }
+
+    setPodName(): void {
+        // in a pod the hostname is the pod name
+        const host = os.hostname();
+        if (host.startsWith('ts-wkr-')) {
+            this.default_labels.pod_name = host;
+        }
+    }
+
+    /**
+     * [set metric to value]
+     * @param  {string} name [metric name]
+     * @param  {Array<string>} labelValues [list of label values]
+     * @param  {number} value [metric value]
+     * @return {Promise<void>}
+     */
+    async set(name: string, labelValues: Array<string>, value: number): Promise<void> {
+        if (this.metricList[name].functions.has('set')) {
+            this.metricList[name].metric.labels(...labelValues.concat(
+                Object.values(this.default_labels)
+            )).set(value);
+        }
+    }
+
+    /**
+     * [increment metric by value]
+     * @param  {string} name [metric name]
+     * @param  {Array<string>} labelValues [list of label values]
+     * @param  {number} value [metric value, default = 1]
+     * @return {Promise<void>}
+     */
+    async inc(name: string, labelValues: Array<string>, value = 1): Promise<void> {
+        if (this.metricList[name].functions.has('inc')) {
+            this.metricList[name].metric.labels(...labelValues.concat(
+                Object.values(this.default_labels)
+            )).inc(value);
+        }
+    }
+
+    /**
+     * [decrement metric by value]
+     * @param  {string} name [metric name]
+     * @param  {Array<string>} labelValues [list of label values]
+     * @param  {number} value [metric value, default = 1]
+     * @return {Promise<void>}
+     */
+    async dec(name: string, labelValues: Array<string>, value = 1): Promise<void> {
+        if (this.metricList[name].functions.has('dec')) {
+            this.metricList[name].metric.labels(...labelValues.concat(
+                Object.values(this.default_labels)
+            )).dec(value);
+        }
+    }
+    /**
+     * [observe value, used by summary metric type]
+     * @param  {string} name [metric name]
+     * @param  {Array<string>} labelValues [list of label values]
+     * @param  {number} value [metric value]
+     * @return {Promise<void>}
+     */
+    async observe(name: string, labelValues: Array<string>, value: number): Promise<void> {
+        if (this.metricList[name].functions.has('observe')) {
+            this.metricList[name].metric.labels(...labelValues.concat(
+                Object.values(this.default_labels)
+            )).observe(value);
+        }
+    }
+
+    /**
+     * [add (define) new metric]
+     * @param  {string} name [metric name]
+     * @param  {string} help [metric help]
+     * @param  {Array<string>} labelsNames [list of label names]
+     * @param  {string} type [metric type, gauge, counter, histogram, or summary]
+     * @return {Promise<void>}
+     */
+    async add(name: string, help: string, labelsNames: Array<string>,
```

Comment:
The name here is hard to distinguish from `inc` and `dec` since `add` is a synonym to `inc`. Renaming this to `addMetric` would be more intuitive. Also since this method is called `addMetric`, I would expect it to throw if the metric is already defined, since this would be prevent accidental or non-obvious user problems, however in order to do that we need two new methods, one to check whether the metric already exists (`hasMetric(name: string)`) and one to remove an existing one `deleteMetric(name: string)`.